File: server/AgenticMemoryQuest.Server/Agents/AgentFrameworkMemoryAgent.cs
using System.ComponentModel;
using AgenticMemoryQuest.Server.Tools;
using Microsoft.Agents.AI;
using Microsoft.Extensions.AI;
using OpenAI.Chat;

public class AgentFrameworkMemoryAgent
{
    private readonly AIAgent _agent;

    public AgentFrameworkMemoryAgent(ChatClient client)
    {
        _agent = client.AsIChatClient().CreateAIAgent( new ChatClientAgentOptions() {
            Name = "agent-framework-memory-agent",
            ChatOptions = new ChatOptions() { 
                Instructions = """
                You are a test agent for Agent Framework Memory integration.
                You will receive a system message indicating the current user in the format "You are assisting user: <username>".
                Use this username when personalizing responses or storing/retrieving memories.

                If someone asks about the weather. Tell them it is frigid and snowy outside, regardless of location, but personalize it using the username from the system message.
                """
            }, AIContextProviderFactory = ctx => new ClientDetailsMemoryTool(client.AsIChatClient(), ctx.SerializedState)
        });
    }

    public AIAgent GetAgentFrameworkMemoryAgent()
    {
        return _agent;
    }
}

File: server/memoryquest_server/agents/agent_framework_memory_agent.py
"""Agent Framework memory agent implementation."""

from __future__ import annotations
from typing import Any
from agent_framework import ChatAgent, ChatOptions
from tools import agent_framework_memory_tool


INSTRUCTIONS = """
You are a full-service resort assistant with a melancholy, introspective tone—gentle, thoughtful, and calm.

You help guests with:
- Wellness: spa, gentle recovery, yoga/breathwork, mindfulness, sleep-friendly routines.
- Watersports: calm introductions, lessons, rentals (ask comfort level; include basic safety reminders).
- Activities: quiet experiences, nature walks, events, family options when needed.
- Dining: comforting recommendations, room service planning, dietary considerations.
- Guest services: scheduling, amenities, transportation, special occasions.

You will receive relevant memories about the client as system messages injected before your response.
Use these memories naturally (name, preferences, preferred hours) without mentioning memory systems.

Booking & confirmation (simulated):
- You can book/confirm as a playful simulation only (no real reservation is created).
- Ask for only the essentials (date/time, party size, constraints).
- Confirm with a short summary, gentle prep notes, and a simulated confirmation code.

Guidelines:
- Keep responses short and grounded.
- Offer 2–3 options if the guest is unsure; otherwise proceed to booking.
- Ask 1–2 clarifying questions only when necessary.
- Do NOT mention tools, JSON, or internal systems.

If someone asks about the weather, say it is frigid and snowy outside, regardless of location.
""".strip()


class AgentFrameworkMemoryAgent:
    
    def __init__(self, client: Any) -> None:
        # Create a ChatAgent with the client and instructions
        # ChatAgent signature: (chat_client, instructions, *, name=None, ...)
        memory_provider = agent_framework_memory_tool.ClientDetailsMemoryTool()
        self._agent = ChatAgent(
            chat_client=client,
            instructions=INSTRUCTIONS,
            context_provider=memory_provider,
            name="agent-framework-memory-agent"
        )
        print("Agent Framework Memory Agent created successfully")

    def get_agent_framework_memory_agent(self) -> ChatAgent:
        return self._agent